test/blackbox-tests/test-cases/github10721.t

```diff
@@ -0,0 +1,21 @@
+
+  $ cat > dune-project << EOF
+  > (lang dune 3.0)
+  > EOF
+
+  $ cat > dune-workspace << EOF
+  > (lang dune 3.0)
+  > (context
+  >   (opam
+  >     (switch /absolute/path/to/opam/switch)))
+  > EOF
+
+  $ dune build
+  File "dune-workspace", line 4, characters 12-41:
+  4 |     (switch /absolute/path/to/opam/switch)))
+                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  Error: The generated context name "/absolute/path/to/opam/switch" from the
+  switch name is invalid.
```

Comment:
How about: this switch name can not be used as a context name?

Reply:
I think this is the same issue we have in https://github.com/ocaml/dune/pull/10723#discussion_r1678073928: users won't understand why Dune is talking about `context  name` whereas we never express a context name. The information that it generated and that we need a `(name ...)` to prevent the automatic generation should be mandatory IMO.

Reply:
Ok, then how about something like this:

> This switch name can not be used as a context name. Please use (name) to disambiguate
> Hint: (name xxx) would be a valid context name

Something like that.

My points are:
- we don't need to repeat the switch name because it's pointed by the loc in the error message
- It's clear that we're defining a context. We're in a `(context)` field.

Reply:
I agree with this message, but I would still emphases the fact it is auto generated. I would add:

> The name generated from this switch can not be used as a context name. Please use (name) to disambiguate
> Hint: (name xxx) would be a valid context name

WDYT?

I agree on both of your points :+1:

Reply:
Yes, that would work